stackit/internal/services/iaas/network/resource.go

```diff
@@ -450,59 +632,149 @@ func toCreatePayload(ctx context.Context, model *Model) (*iaas.CreateNetworkPayl
+func toUpdatePayload(ctx context.Context, model, stateModel *Model) (*iaas.PartialUpdateNetworkPayload, error) {
 	if model == nil {
 		return nil, fmt.Errorf("nil model")
 	}
+	addressFamily := &iaas.UpdateNetworkAddressFamily{}
 
-	modelNameservers := []string{}
-	for _, ns := range model.Nameservers.Elements() {
-		nameserverString, ok := ns.(types.String)
+	modelIPv6Nameservers := []string{}
+	for _, ipv6ns := range model.IPv6Nameservers.Elements() {
+		ipv6NameserverString, ok := ipv6ns.(types.String)
 		if !ok {
 			return nil, fmt.Errorf("type assertion failed")
 		}
-		modelNameservers = append(modelNameservers, nameserverString.ValueString())
+		modelIPv6Nameservers = append(modelIPv6Nameservers, ipv6NameserverString.ValueString())
+	}
+
+	if !model.IPv6Nameservers.IsNull() && len(model.IPv6Nameservers.Elements()) > 0 {
+		addressFamily.Ipv6 = &iaas.UpdateNetworkIPv6Body{
+			Nameservers: &modelIPv6Nameservers,
+		}
+
+		if model.NoIPv6Gateway.ValueBool() {
+			addressFamily.Ipv6.Gateway = iaas.NewNullableString(nil)
+		} else if !(model.IPv6Gateway.IsUnknown() || model.IPv6Gateway.IsNull()) {
+			addressFamily.Ipv6.Gateway = iaas.NewNullableString(conversion.StringValueToPointer(model.IPv6Gateway))
+		}
 	}
 
+	modelIPv4Nameservers := []string{}
+	var modelIPv4List []attr.Value
+
+	if !(model.IPv4Nameservers.IsNull() || model.IPv4Nameservers.IsUnknown()) {
+		modelIPv4List = model.IPv4Nameservers.Elements()
+	} else {
+		modelIPv4List = model.Nameservers.Elements()
+	}
+	for _, ipv4ns := range modelIPv4List {
+		ipv4NameserverString, ok := ipv4ns.(types.String)
+		if !ok {
+			return nil, fmt.Errorf("type assertion failed")
+		}
+		modelIPv4Nameservers = append(modelIPv4Nameservers, ipv4NameserverString.ValueString())
+	}
+
+	if !model.IPv4Nameservers.IsNull() || !model.Nameservers.IsNull() {
+		addressFamily.Ipv4 = &iaas.UpdateNetworkIPv4Body{
+			Nameservers: &modelIPv4Nameservers,
+		}
+	}
+
+	if model.NoIPv4Gateway.ValueBool() {
+		addressFamily.Ipv4.Gateway = iaas.NewNullableString(nil)
+	} else if !(model.IPv4Gateway.IsUnknown() || model.IPv4Gateway.IsNull()) {
+		addressFamily.Ipv4.Gateway = iaas.NewNullableString(conversion.StringValueToPointer(model.IPv4Gateway))
+	}
+
+	currentLabels := stateModel.Labels
 	labels, err := conversion.ToJSONMapPartialUpdatePayload(ctx, currentLabels, model.Labels)
 	if err != nil {
 		return nil, fmt.Errorf("converting to Go map: %w", err)
 	}
 
-	return &iaas.PartialUpdateNetworkPayload{
-		Name: conversion.StringValueToPointer(model.Name),
-		AddressFamily: &iaas.UpdateNetworkAddressFamily{
-			Ipv4: &iaas.UpdateNetworkIPv4Body{
-				Nameservers: &modelNameservers,
-			},
-		},
+	payload := iaas.PartialUpdateNetworkPayload{
+		Name:   conversion.StringValueToPointer(model.Name),
 		Labels: &labels,
-	}, nil
+	}
+
+	if addressFamily.Ipv6 != nil || addressFamily.Ipv4 != nil {
+		payload.AddressFamily = addressFamily
+	}
+
+	return &payload, nil
```

Comment:
Why is routed never set in update payloads?

Reply:
This leads to an inconsistent result after apply when we try to change it. If we don't want to allow updates in the `routed` field for some reason, we should make it `RequiresReplace`

Reply:
Routed is not yet supported in update operations by API. That is why we have decided to remove it from the payloads. 

Thx for the hint, I added `REquiresReplace`